### PR TITLE
[HttpKernel] 15874 framework exceptions

### LINKED ISSUE #15874: [HttpKernel] Add support for configuring log level, and status code by …
| Q            | A
| ------------ | ---
| Feature PR   | symfony/symfony#42244
| PR author(s) | @lyrixx
| Merged in    | 5.4

### PR DESCRIPTION
**Pull request by JohJohan**
Fixes #15874

### CODE REVIEW

File: reference/configuration/framework.rst [hunk 1]
```diff
@@ -3316,6 +3316,64 @@ Defines the kind of workflow that is going to be created, which can be either
 a normal workflow or a state machine. Read :doc:`this article </workflow/workflow-and-state-machine>`
 to know their differences.
 
+.. _reference-workflows-exceptions:
+
+exceptions
+""""
```
Comment (lyrixx) [inline reference/configuration/framework.rst]:
```suggestion
exceptions
""""""""""
```

File: reference/configuration/framework.rst [hunk 2]
```diff
@@ -3316,6 +3316,64 @@ Defines the kind of workflow that is going to be created, which can be either
 a normal workflow or a state machine. Read :doc:`this article </workflow/workflow-and-state-machine>`
 to know their differences.
 
+.. _reference-workflows-exceptions:
```
Comment (lyrixx) [inline reference/configuration/framework.rst]: it looks like it's a bad copy/paste
And if we don't reference this section, I think you can drop this line

**Review by lyrixx — COMMENTED** [review]
Thanks. I let few comments

**Comment by JohJohan** [comment]
> Thanks. I let few comments

Thanks for the review, should i make it one commit again? Or is it okay like this?

**Review by lyrixx — APPROVED** [review]

**Comment by kbond** [comment]
I was wondering if it uses an instance of the exception class. I checked the code and it does, maybe this should be documented? Also, [the order is important](https://github.com/symfony/symfony/pull/42244/files#r675956867).

**Comment by JohJohan** [comment]
> I was wondering if it uses an instance of the exception class. I checked the code and it does, maybe this should be documented? Also, [the order is important](https://github.com/symfony/symfony/pull/42244/files#r675956867).

@kbond So it would be nice to have 2 notices 1 mentioning the exception should extend from `\Exception`? And one note about that the ordering is imported?

**Comment by kbond** [comment]
> mentioning the exception should extend from \Exception

This isn't what I meant. I meant that the first exception class that passes an `instanceof` is what is used.

I think one notice is sufficient as it is related:

```yaml
framework:
    exceptions:
        Exception:
            log_level: debug
            status_code: 404
        RuntimeException: # this will never be used as \RuntimeException extends \Exception
            log_level: debug
            status_code: 422
```

The fix would be to reverse the above:

```yaml
framework:
    exceptions:
        RuntimeException:
            log_level: debug
            status_code: 422
        Exception:
            log_level: debug
            status_code: 404
```

Hope that makes sense.

**Review by lyrixx — APPROVED** [review]

**Comment by javiereguiluz** [comment]
Thanks Johan! This is now merged.

**Pull request merged**